src/vs/editor/common/services/getIconClasses.ts

```diff
@@ -26,6 +27,9 @@ export function getIconClasses(modelService: IModelService, modeService: IModeSe
 			name = cssEscape(basenameOrAuthority(resource).toLowerCase());
 		}
 
+		// Directory
+		classes.push(`${Paths.basename(Paths.dirname(resource.path))}-name-dir-icon`);
```

Comment:
I think we need `cssEscape` and lowercase the folder name.

Reply:
Resolved in 77503a9.